tests/integration/utils/random.py

```diff
@@ -154,12 +154,18 @@ def assert_distribution(a, theo_mean, theo_stdev, tolerance=1e-2):
             num.mean((a - average) ** 2)
         )  # num.std(a) -> does not work
     print(
-        f"average = {average} - expected {theo_mean}"
-        + f", stdev = {stdev} - expected {theo_stdev}\n"
+        f"average = {average} - theoretical {theo_mean}"
+        + f", stdev = {stdev} - theoretical {theo_stdev}\n"
     )
-    assert np.abs(theo_mean - average) < tolerance * np.max(
+    assert np.abs(theo_mean - average) < mean_tol * np.max(
         (1.0, np.abs(theo_mean))
     )
-    assert np.abs(theo_stdev - stdev) < tolerance * np.max(
-        (1.0, np.abs(theo_stdev))
-    )
+    # the theoretical standard deviation can't be 0
+    assert theo_stdev != 0
+    # TODO: this check is not a good proxy to validating that the samples
+    #       respect the assumed random distribution unless we draw
+    #       extremely many samples. until we find a better validation
+    #       method, we make the check lenient to avoid random
+    #       failures in the CI. (we still need the check to catch
+    #       the cases that are obviously wrong.)
+    assert np.abs(theo_stdev - stdev) < stdev_tol * np.abs(theo_stdev)
```

Comment:
The default `stddev_tol = 2 = 200%` seems like a very high limit for a relative difference. Since standard deviation is always non-negative, what you are essentially checking here is that `stddev < 3 * theo_stdev`. Is that your goal here, or is `stddev_tol = 2` a typo?

Reply:
Oops. I spent much time on writing a comment and didn't do the job for the test itself ;) I revised the test to make it more sensible.